### PR TITLE
nas-cs01z Support

### PR DESCRIPTION
**Pull request by Arty68**

### CODE REVIEW

File: config/shenzen_neo/nas-cs01z.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Product xmlns='https://github.com/OpenZWave/open-zwave' Revision="1">
+    <!-- NAS-CS01Z
```
Comment (Fishwaldo) [inline config/shenzen_neo/nas-cs01z.xml:3]: Please add MetaData Block
https://github.com/OpenZWave/open-zwave/wiki/Adding-Devices#creating-the-metadata-block

**Comment by Fishwaldo** [comment]
Ping?

**Comment by Arty68** [comment]
> Ping?

Hi I will do it but no time for now.

**Comment by Fishwaldo** [comment]
Please reopen when you have completed this.

**Comment by Arty68** [comment]
Hi I need to add a new device here http://www.openzwave.com/device-database/ how can I do ?
Thanks

File: config/shenzen_neo/nas-cs01z.xml (hunk 1 above)
Comment (jjnh) [reply config/shenzen_neo/nas-cs01z.xml:3]: Hi, trying to build the metadata file, but I cannot find the device on the Z-Wave alliance website. However I found a complete description of it here: https://www.cd-jackson.com/index.php/zwave/zwave-device-database/zwave-device-list/devicesummary/1108

Comment (jjnh) [reply config/shenzen_neo/nas-cs01z.xml:3]: `<MetaData>
	<MetaDataItem name="Name">NEO Coolcam Roller Shutter</MetaDataItem>
	<MetaDataItem name="Description">NEO Coolcam Roller Shutter - Curtain Switch - NAS-SC03ZE-2-T-V3</MetaDataItem>
	<MetaDataItem name="OzwInfoPage">http://www.openzwave.com/device-database/0258:102c:0200</MetaDataItem>
	<MetaDataItem name="ProductPage">https://szneo.com/en/products/show.php?id=234</MetaDataItem>
	<MetaDataItem name="ProductSupport">https://szneo.com/en/service/index.php</MetaDataItem>
	<MetaDataItem name="ProductPic">https://www.cd-jackson.com/zwave_device_uploads/1108/neo-curtain-Switch.JPG</MetaDataItem>
	<MetaDataItem name="ProductManual">https://www.cd-jackson.com/zwave_device_uploads/1108/NeoRollerS.pdf</MetaDataItem>
	<MetaDataItem name="WakeupDescription">Power on the device</MetaDataItem>
	<MetaDataItem name="InclusionDescription">1. Set Zwave controller intro inclusion mode
2. power on the device
3. touch the middle button 3 times within 1s
4. the device will be entered into inclusion mode automatically
(All Led lights will be bliked with 1s interval until inclusion successfull)</MetaDataItem>
	<MetaDataItem name="ExclusionDescription">1. Set Z-WaveControll into exclusion mode
2. Power on the device
3. Touch the middle button 3 times within 1s
4. the device will be entered into exclusion mode
(All led lights will de blinked 3 times with 0.5s interval.)</MetaDataItem>
	<MetaDataItem name="ResetDescription">Power on the device
	Keep touching the middle button for 10s until the led turns to red, then release</MetaDataItem>
	<MetaDataItem id="102C" name="ZWProductPage" type="0200">Unknown</MetaDataItem>
	<MetaDataItem id="102C" name="FrequencyName" type="0200">CEPT (Europe)</MetaDataItem>
	<MetaDataItem id="102C" name="Identifier" type="0200">NAS-SC03ZE-2-T-V3</MetaDataItem>
	<ChangeLog>
		<Entry author="John Huppertz - jhuppertz@ziggo.nl" date="02 December 2020" revision="1">Initial setup of metadata file</Entry>
	</ChangeLog>
</MetaData>`

**Comment by jjnh** [comment]
Hi, in the meantime I have been working through some XML files and amended the config with the new switch. I'm no programmer so it mostly trial and error. At the moment I still have the shutter switch as "unknown" in the device list (using Domoticz). I read that I need to remove several XML files (config & cache). Will I lose my running config if I do that?

**Comment by jjnh** [comment]
Ok, I got it working using several files from forums.

I added one line to the "manufacturer_specific.xml" in the Shenzhen Neo Electronics Co Ltd (0258) part:

`<Product` config="shenzen_neo/nas-sc03ze.xml" id="102c" name="NAS-SC03ZE Roller Shutter" type="0200"/>`

The I created the attached "nas-sc03ze.xml" file.
[nas-sc03ze.xml.txt](https://github.com/OpenZWave/open-zwave/files/5648765/nas-sc03ze.xml.txt)

This should work for the Neo Coolcam NAS-SC01ZE and NAS-SC03ZE. I tested it successful with the NAS-SC03ZE.

**Comment by Arty68** [comment]
Hi nice to see that someone has done something with this device.
Are those devices working good ?
By me some are working good but I have 4 devices which are still losing there configuration after one month.

See you
Arty

**Comment by jjnh** [comment]
Thanks good to hear that I am not the only one :-)

So far I only have one is use for testing as we are expecting new windows with electric shutters and screens somewhere beginning of next year. The one I have is working good so far (1 week of usage is pretty short).  It has not lost it's configuration ... yet. I have seen other reports from users having this issue but 4 is a large number. Seems like you caught a bad batch. How many do you have in use in total?

Cheers,
John

**Comment by jjnh** [comment]
Well it started losing it's configuration now. Every 2 to 5 days I have to completely reprogram it. Calibration and Z-Wave connection are completely lost. Reprogramming works, but a few days later again the led is blinking and the config is gone. I ordered a Zigbee switch to test if that works stable.

**Comment by Arty68** [comment]
one is stable ( no configuration lose for a year, the others are not very stable conf lose every month more or less) but I wonder if this is not linked with domoticz.
I will try to use home assistant but it's a bit boring because I will have to migrate my script also :S

**Comment by jjnh** [comment]
Since the last config loss, I have recalibrated the switch, but did not reconnect it to Domoticz. Now it retained it's calibration for more then a month. 
I suspect that somehow Openzwave causes the loss of the calibration and connection.

**Comment by Arty68** [comment]
> Since the last config loss, I have recalibrated the switch, but did not reconnect it to Domoticz. Now it retained it's calibration for more then a month.
> I suspect that somehow Openzwave causes the loss of the calibration and connection.

It's also what I am suspecting. Either domoticz or openzwave.

**Comment by Arty68** [comment]
After some time I got the same issues again. It's boring 😠

**Pull request closed without merging**